File: GTTS/INL.TrainingService.Models/IGetTrainingEvent_Item.cs
using System;
using System.Collections.Generic;

namespace INL.TrainingService.Models
{
    public interface IGetTrainingEvent_Item
    {
        long TrainingEventID { get; set; }
        string Name { get; set; }
        string NameInLocalLang { get; set; }
        int TrainingEventTypeID { get; set; }
        string TrainingEventTypeName { get; set; }
        string Justification { get; set; }
        string Objectives { get; set; }
        string ParticipantProfile { get; set; }
        string SpecialRequirements { get; set; }
        int? ProgramID { get; set; }
        int? KeyActivityID { get; set; }
        string KeyActivityName { get; set; }
        long? TrainingUnitID { get; set; }
        string BusinessUnitAcronym { get; set; }
        string BusinessUnitName { get; set; }
        int? CountryID { get; set; }
        int? PostID { get; set; }
        int? ConsularDistrictID { get; set; }
        int? OrganizerAppUserID { get; set; }
        int? PlannedParticipantCnt { get; set; }
        int? PlannedMissionDirectHireCnt { get; set; }
        int? PlannedNonMissionDirectHireCnt { get; set; }
        int? PlannedMissionOutsourceCnt { get; set; }
        int? PlannedOtherCnt { get; set; }
        decimal? EstimatedBudget { get; set; }
        decimal? ActualBudget { get; set; }
        int? EstimatedStudents { get; set; }
        int? FundingSourceID { get; set; }
        int? AuthorizingLawID { get; set; }
        string Comments { get; set; }
        int? ModifiedByAppUserID { get; set; }
        DateTime? ModifiedDate { get; set; }
        DateTime? TravelStartDate { get; set; }
        DateTime? TravelEndDate { get; set; }
        DateTime? EventStartDate { get; set; }
        DateTime? EventEndDate { get; set; }
        List<ParticipantCount_Item> ParticipantCounts { get; set; }
        int? TrainingEventStatusID { get; set; }
        string TrainingEventStatus { get; set; }
        GetTrainingEventAppUser_Item Organizer { get; set; }
        GetTrainingEventAppUser_Item ModifiedBy { get; set; }

        List<GetTrainingEventLocation_Item> TrainingEventLocations { get; set; }
        List<GetTrainingEventUSPartnerAgency_Item> TrainingEventUSPartnerAgencies { get; set; }
        List<GetTrainingEventProjectCode_Item> TrainingEventProjectCodes { get; set; }
        List<GetTrainingEventStakeholder_Item> TrainingEventStakeholders { get; set; }
        List<GetTrainingEventAttachment_Item> TrainingEventAttachments { get; set; }
        List<GetTrainingCourseDefinitionProgram_Item> TrainingEventCourseDefinitionPrograms { get; set; }
        List<GetTrainingEventKeyActivity_Item> TrainingEventKeyActivities { get; set; }
        List<IAA_Item> IAAs { get; set; }
    }
}

File: GTTS/GTTS.Web/ClientApp/src/app/models/INL.TrainingService.Models/iget-training-event_item.ts



import { ParticipantCount_Item } from './participant-count_item';
import { GetTrainingEventAppUser_Item } from './get-training-event-app-user_item';
import { GetTrainingEventLocation_Item } from './get-training-event-location_item';
import { GetTrainingEventUSPartnerAgency_Item } from './get-training-event-uspartner-agency_item';
import { GetTrainingEventProjectCode_Item } from './get-training-event-project-code_item';
import { GetTrainingEventStakeholder_Item } from './get-training-event-stakeholder_item';
import { GetTrainingEventAttachment_Item } from './get-training-event-attachment_item';
import { GetTrainingCourseDefinitionProgram_Item } from './get-training-course-definition-program_item';
import { GetTrainingEventKeyActivity_Item } from './get-training-event-key-activity_item';
import { IAA_Item } from './iaa_item';

export interface IGetTrainingEvent_Item {
  
	TrainingEventID: number;
	Name: string;
	NameInLocalLang: string;
	TrainingEventTypeID: number;
	TrainingEventTypeName: string;
	Justification: string;
	Objectives: string;
	ParticipantProfile: string;
	SpecialRequirements: string;
	ProgramID?: number;
	KeyActivityID?: number;
	KeyActivityName: string;
	TrainingUnitID?: number;
	BusinessUnitAcronym: string;
	BusinessUnitName: string;
	CountryID?: number;
	PostID?: number;
	ConsularDistrictID?: number;
	OrganizerAppUserID?: number;
	PlannedParticipantCnt?: number;
	PlannedMissionDirectHireCnt?: number;
	PlannedNonMissionDirectHireCnt?: number;
	PlannedMissionOutsourceCnt?: number;
	PlannedOtherCnt?: number;
	EstimatedBudget?: number;
	ActualBudget?: number;
	EstimatedStudents?: number;
	FundingSourceID?: number;
	AuthorizingLawID?: number;
	Comments: string;
	ModifiedByAppUserID?: number;
	ModifiedDate?: Date;
	TravelStartDate?: Date;
	TravelEndDate?: Date;
	EventStartDate?: Date;
	EventEndDate?: Date;
	ParticipantCounts?: ParticipantCount_Item[];
	TrainingEventStatusID?: number;
	TrainingEventStatus: string;
	Organizer?: GetTrainingEventAppUser_Item;
	ModifiedBy?: GetTrainingEventAppUser_Item;
	TrainingEventLocations?: GetTrainingEventLocation_Item[];
	TrainingEventUSPartnerAgencies?: GetTrainingEventUSPartnerAgency_Item[];
	TrainingEventProjectCodes?: GetTrainingEventProjectCode_Item[];
	TrainingEventStakeholders?: GetTrainingEventStakeholder_Item[];
	TrainingEventAttachments?: GetTrainingEventAttachment_Item[];
	TrainingEventCourseDefinitionPrograms?: GetTrainingCourseDefinitionProgram_Item[];
	TrainingEventKeyActivities?: GetTrainingEventKeyActivity_Item[];
	IAAs?: IAA_Item[];
}





